tests-dev/ufs_test.sh

```diff
@@ -0,0 +1,267 @@
+	export UFS_TEST_YAML
+	;;
+    c)
+	CREATE_BASELINE=true
+	;;
+    l)
+	TESTS_FILE=${OPTARG}
+	grep -q '[^[:space:]]' < "${TESTS_FILE}" ||  die "${TESTS_FILE} empty, exiting..."
+	UFS_TEST_YAML=${TESTS_FILE}
+	export UFS_TEST_YAML
+	;;
+    o)
+	COMPILE_ONLY=true
+	;;
+    m)
+	# redefine RTPWD to point to newly created baseline outputs
+	RTPWD_NEW_BASELINE=true
+	;;
+    n)
+	RUN_SINGLE_TEST=true
+	IFS=' ' read -r -a SINGLE_OPTS <<< "${OPTARG}"
+
+	if [[ ${#SINGLE_OPTS[@]} != 2 ]]; then
+            die 'The -n option needs <testname> AND <compiler> in quotes, i.e. -n "control_p8 intel"'
+	fi
+
+	SRT_NAME="${SINGLE_OPTS[0]}"
+	SRT_COMPILER="${SINGLE_OPTS[1]}"
+
+	if [[ "${SRT_COMPILER}" != "intel" ]] && [[ "${SRT_COMPILER}" != "gnu" ]]; then
+            die "COMPILER MUST BE 'intel' OR 'gnu'"
+	fi
+
+	export SRT_NAME
+	export SRT_COMPILER
+	python -c "import create_yml; create_yml.update_testyaml_n()"
+	UFS_TEST_YAML="ufs_test_temp.yaml"
+	export UFS_TEST_YAML
+	;;
+    d)
+	export delete_rundir=true
+	;;
+    w)
+	export skip_check_results=true
+	;;
+    k)
+	KEEP_RUNDIR=true
+	;;
+    r)
+	ROCOTO=true
+	ECFLOW=false
+	;;
+    e)
+	ECFLOW=true
+	ROCOTO=false
+	die "Work-in-progress to support for ECFLOW. Please, use the ROCOTO workflow manamegment option (-r)"
+	;;
+    s)
+	LINK_TESTS=true
+	;;
+    h)
+	usage
+	;;
+    :)
+	die "Option -${OPTARG} requires an argument."
+	;;
+    ?)
+	die "Invalid option: -${OPTARG}"
+	;;
+    *)
+        die "Invalid runtime options: no parameter included with argument-${OPTARG}"
+	;;
+  esac
+done
+
+# If -s; link sharable test scripts from tests directory
+if [[ ${LINK_TESTS} == true ]]; then
+    python -c "import ufs_test_utils; ufs_test_utils.sync_testscripts()"
+fi
+
+source detect_machine.sh # Note: this does not set ACCNR. The "if" block below does.
+source rt_utils.sh
+source module-setup.sh
+
+#Check to error out if incompatible options are chosen together
+[[ ${KEEP_RUNDIR} == true && ${delete_rundir} == true ]] && die "-k and -d options cannot be used at the same time"
+[[ ${ECFLOW} == true && ${ROCOTO} == true ]] && die "-r and -e options cannot be used at the same time"
+[[ ${CREATE_BASELINE} == true && ${RTPWD_NEW_BASELINE} == true ]] && die "-c and -m options cannot be used at the same time"
+
+if [[ -z "${ACCNR}" ]]; then
+  echo "Please use -a <account> to set group account to use on HPC"
+  exit 1
+fi
+
+# Display the machine and account using the format detect_machine.sh used:
+echo "Machine:  ""${MACHINE_ID}""    Account: ""${ACCNR}"" "
+
+check_machine=false
+platforms=( hera orion hercules gaea jet derecho noaacloud s4 )
+for name in "${platforms[@]}"
+do
+  if [[ ${MACHINE_ID} == "${name}" ]]; then
+    check_machine=true
+    break
+  fi
+done
+
+if [[ ${check_machine} == true ]]; then
+    source "${PATHRT}"/machine_config/machine_"${MACHINE_ID}".config
+else
+    die "*** Current support of ufs_test.sh only for hera orion hercules gaea jet derecho noaacloud s4 ! ***"
```

Comment:
Use platforms variable not to duplicate code

Reply:
Do you think another machine variable other than ${MACHINE_ID}?

Reply:
This is more minor. I understand that the else statement repeats the machines listed in the platforms variable. Using the platforms variable avoids duplicate code for what machines are supported.

Reply:
update with  ${platforms[*]